lance-spark-base_2.12/src/main/antlr4/org/apache/spark/sql/catalyst/parser/extensions/LanceSqlExtensions.g4

```diff
@@ -19,7 +19,8 @@ singleStatement
     ;
 
 statement
-    : ALTER TABLE multipartIdentifier ADD COLUMNS columnList FROM identifier        #addColumnsBackfill
+    : ALTER TABLE multipartIdentifier ADD COLUMNS columnList FROM identifier                            #addColumnsBackfill
+    | ALTER TABLE multipartIdentifier COMPACT (WITH '(' (namedArgument (',' namedArgument)*)? ')')?     #compact
```

Comment:
I think for compaction we should use the syntax similar to Delta: https://docs.databricks.com/aws/en/sql/language-manual/delta-optimize

Given our roadmap to also support features like liquid clustering and z-order compaction.